content/authoring/kata/create-first-kata.md

```diff
@@ -0,0 +1,266 @@
+
+For normal kata, a good set of tests should cover all of these aspects:
+- Test basic functionality
+- Has full coverage (if that's impossible, at least have decent coverage)
+- Cover edge cases thoroughly
+- Randomized tests to probe user solution with random samples (and so that pattern-matching against the tests is impossible)
+- Stress/performance/code characteristic tests if needed
+
+The first three should be put into fixed tests. The fourth item should be put into random tests (see below). Ideally the last item would be in isolation, or covered by random tests. The last item is optional.
+
+Random test cases are test cases that cannot be predicted. Most kata have them (except for the really old ones) and they are usually in addition to some static tests. Using both static and random test cases with make it both easy for warriors to see what they are supposed to do, as well as make it impossible to write a solution that just pattern match the inputs (i.e return hard-coded outputs depending on a fixed set of inputs). Random tests are also good at finding edge cases.
+
+Remember: just like in real life, if we failed a test, we want to know:
+- what was the input
+- expected and actual result
+
+So unless revealing the expected result would spoil the kata, you should not hide them. Consult the documentation of testing framework used by your [language](/languages/) and pick the best method for your tests.
+
+Writing good test suites is very difficult, often the most difficult part of creatign a kata. It often requires more code than the actual solution, and any flaws in the test suite can heavily impact the overall quality and experience of users with the kata. Make sure you follow [quality guidelines for submssion tests](/authoring/guidelines/submission-tests/) when writing them, as they will help you to avoid many common pitfalls awaiting for inexperienced authors.
+
+
+#### Example Test Cases
+
+Example test cases is a small set of tests which user can see and modify while working on their solution. These are some basic test cases that users will see when they load the kata. Sample tests are written in the same way as submission tests, using the testing framework set up for your language.
+
+They are not necessary, but they are really good to have, so users can get the idea how the solution is called and tested.You can include a few tests to get someone started, though of course if you're lazy you can just copy the fixed test cases from the actual tests over there. Just don't copy your reference solution there as well. Unless you intend the users to write tests themselves, or tey are not applicable for your kata (Example: [Defuse the bombs](https://www.codewars.com/kata/defuse-the-bombs/)), it's usually considered a good practice to provide example test cases to the users.
```

Comment:
```suggestion
Except in circumstances where providing sample tests to the user would spoil the kata (such as [Defuse the bombs](https://www.codewars.com/kata/defuse-the-bombs/)), they are absolutely **required** as users can get an idea of how the solution is called and tested. You should include a few tests to get someone started, though of course if you're lazy you can just copy over the fixed assertions in your submission tests to serve as the sample tests. Just remember *not* to copy your reference solution there as well. **It is a Kata issue if there are no sample tests** unless strong justification can be provided against them for a particular Kata.
```

Reply:
I am going to merge this one to get fixes for wording etc. but it's to be reviewed.  
I think it's been agreed that sample tests are _not_ "absolutely required" and other places of docs mention this. I will check and make sure that two (or more) docs do not contradict each other.